### PR TITLE
Misc CI cleanups

### PR DESCRIPTION
**Pull request by njsmith**
- Update branch names to match upstream
- Drop py2 testing
- Appease stricter checks in latest Sphinx version

Split off from #1026 as requested.

### CODE REVIEW

**Comment by alex** [comment]
Looks like CI is failing. Is this related to this PR or was it pre-broken?

**Comment by njsmith** [comment]
Oh it was pre-broken, but I think I just pushed the piece to unbreak it while you were commenting

File: tox.ini [hunk 1]
```diff
@@ -9,8 +9,7 @@ extras =
     test
 deps =
     coverage>=4.2
-    cryptographyMaster: git+https://github.com/pyca/cryptography.git
-    cryptographyMinimum: cryptography==3.3
```
Comment (alex) [inline tox.ini:13]: Without this line, how does -minimum work?

Comment (njsmith) [reply tox.ini:13]: Huh, when I told magit to revert that line I wasn't expecting it to delete it entirely. A bit disturbing that nothing breaks with that missing. Anyway, nice catch, thanks :-) And I reverted it properly now...

**Comment by reaperhulk** [comment]
As long as we support that cryptography 3.3 minimum we should keep testing python 2.7, otherwise lgtm.

**Comment by njsmith** [comment]
Well, I don't really care about support for py2 or old cryptographys, but I'm happy to go with whatever.

Does anyone care, or should we just toss them out and move on?

**Review by reaperhulk — APPROVED** [review]

**Pull request merged**